File: ThreeDSecure/src/test/java/com/braintreepayments/api/ThreeDSecureV2TextBoxCustomizationUnitTest.java
package com.braintreepayments.api;

import android.os.Parcel;

import com.cardinalcommerce.shared.userinterfaces.TextBoxCustomization;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;

@RunWith(RobolectricTestRunner.class)
public class ThreeDSecureV2TextBoxCustomizationUnitTest {

    @Test
    public void setsAllCardinalClassProperties() {
        ThreeDSecureV2TextBoxCustomization sut = new ThreeDSecureV2TextBoxCustomization();
        sut.setBorderColor("#FFA500");
        sut.setBorderWidth(1);
        sut.setCornerRadius(7);
        sut.setTextColor("#0000ff");
        sut.setTextFontSize(5);
        sut.setTextFontName("Arial");

        TextBoxCustomization cardinalTextBoxCustomization = sut.getCardinalTextBoxCustomization();
        assertEquals("#FFA500", cardinalTextBoxCustomization.getBorderColor());
        assertEquals(1, cardinalTextBoxCustomization.getBorderWidth());
        assertEquals(7, cardinalTextBoxCustomization.getCornerRadius());
        assertEquals("#0000ff", cardinalTextBoxCustomization.getTextColor());
        assertEquals(5, cardinalTextBoxCustomization.getTextFontSize());
        assertEquals("Arial", cardinalTextBoxCustomization.getTextFontName());
    }

    @Test
    public void writeToParcel() {
        ThreeDSecureV2TextBoxCustomization customization = new ThreeDSecureV2TextBoxCustomization();
        customization.setBorderColor("#FFFFFF");
        customization.setBorderWidth(10);
        customization.setCornerRadius(5);
        customization.setTextColor("#121212");
        customization.setTextFontName("Helvetica");
        customization.setTextFontSize(15);

        Parcel parcel = Parcel.obtain();
        customization.writeToParcel(parcel, 0);
        parcel.setDataPosition(0);

        ThreeDSecureV2TextBoxCustomization actual = (ThreeDSecureV2TextBoxCustomization) ThreeDSecureV2TextBoxCustomization.CREATOR.createFromParcel(parcel);

        assertEquals("#FFFFFF", actual.getBorderColor());
        assertEquals(10, actual.getBorderWidth());
        assertEquals(5, actual.getCornerRadius());
        assertEquals("#121212", actual.getTextColor());
        assertEquals("Helvetica", actual.getTextFontName());
        assertEquals(15, actual.getTextFontSize());
    }
}

File: ThreeDSecure/src/test/java/com/braintreepayments/api/threedsecure/ThreeDSecureV2TextBoxCustomizationUnitTest.kt
package com.braintreepayments.api.threedsecure

import android.os.Parcel
import kotlinx.parcelize.parcelableCreator
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner
import kotlin.test.Test
import kotlin.test.assertEquals

@RunWith(RobolectricTestRunner::class)
class ThreeDSecureV2TextBoxCustomizationUnitTest {

    @Test
    fun `sets all cardinal class properties`() {
        val sut = ThreeDSecureV2TextBoxCustomization(
            borderColor = "#FFA500",
            borderWidth = 1,
            cornerRadius = 7,
            textColor = "#0000ff",
            textFontSize = 5,
            textFontName = "Arial"
        )

        val cardinalTextBoxCustomization = sut.cardinalTextBoxCustomization
        assertEquals("#FFA500", cardinalTextBoxCustomization.borderColor)
        assertEquals(1, cardinalTextBoxCustomization.borderWidth)
        assertEquals(7, cardinalTextBoxCustomization.cornerRadius)
        assertEquals("#0000ff", cardinalTextBoxCustomization.textColor)
        assertEquals(5, cardinalTextBoxCustomization.textFontSize)
        assertEquals("Arial", cardinalTextBoxCustomization.textFontName)
    }

    @Test
    fun `write to parcel`() {
        val sut = ThreeDSecureV2TextBoxCustomization(
            borderColor = "#FFFFFF",
            borderWidth = 10,
            cornerRadius = 5,
            textColor = "#121212",
            textFontName = "Helvetica",
            textFontSize = 15,
        )
        val parcel = Parcel.obtain().apply {
            sut.writeToParcel(this, 0)
            setDataPosition(0)
        }
        val parceled = parcelableCreator<ThreeDSecureV2TextBoxCustomization>().createFromParcel(parcel)

        assertEquals("#FFFFFF", parceled.borderColor)
        assertEquals(10, parceled.borderWidth)
        assertEquals(5, parceled.cornerRadius)
        assertEquals("#121212", parceled.textColor)
        assertEquals("Helvetica", parceled.textFontName)
        assertEquals(15, parceled.textFontSize)
    }
}
